libs/common-traits/src/lib.rs

```diff
@@ -103,26 +106,37 @@ pub trait PoolNAV<PoolId, Amount> {
 	fn update_nav(pool_id: PoolId) -> Result<Amount, DispatchError>;
 }
 
-/// A trait that support pool reserve operations such as borrow and repay
-pub trait PoolReserve<Origin, AccountId> {
+/// Role can hold any type of role specific functions a user can do on a given pool.
+#[derive(Encode, Decode, Clone, Copy, PartialEq)]
+#[cfg_attr(any(feature = "std", feature = "runtime-benchmarks"), derive(Debug))]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub enum Role {
+	PoolAdmin,
+	Borrower,
+	PricingAdmin,
```

Comment:
This has a `PricingAdmin` while the spec doesn't, is that to separate issue from activate/price permissions?

Reply:
yeap. It would better to keep them seperate though